tests/test_readwrite.py

```diff
@@ -368,42 +360,59 @@ def check_compressed(key, value):
     assert_equal(adata, expected)
 
 
-def test_zarr_compression(tmp_path):
-    from numcodecs import Blosc
+@pytest.mark.parametrize("zarr_write_format", [2, 3])
+def test_zarr_compression(tmp_path, zarr_write_format):
+    with ad.settings.override(zarr_write_format=zarr_write_format):
```

Comment:
Instead of adding an indentation level to the whole test, just set the setting.

We should have an autouse fixture resetting it after (and in the unlikely case that we do not, we need to add that!)